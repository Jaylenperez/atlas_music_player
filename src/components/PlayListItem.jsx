export default function PlayListItem() {
  return (
    <div className="flex items-center justify-between">
      <div>
        <h1 className="text-plum font-sans text-lg font-bold">
          Electric Fever
        </h1>
        <p className="text-accent font-sans text-base font-bold">Neon Jungle</p>
      </div>

      <span className="font-sans text-sm text-gray-500">3:45</span>
    </div>
  );
}
