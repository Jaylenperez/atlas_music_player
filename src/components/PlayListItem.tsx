export default function PlayListItem({ title, artist, length }) {
  return (
    <div className="flex items-center justify-between rounded-md border-b border-gray-900 px-4 py-2 hover:bg-indigo-900">
      <div>
        <h3 className="text-lg font-medium text-sky-400">{title}</h3>
        <p className="text-sm text-fuchsia-500">{artist}</p>
      </div>
      <span className="text-sm text-fuchsia-300">{length}</span>
    </div>
  );
}
