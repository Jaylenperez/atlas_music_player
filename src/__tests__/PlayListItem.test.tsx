import { render, screen } from "@testing-library/react";
import { test, expect } from "vitest";
import PlayListItem from "../components/PlayListItem";

test("PlayListItem renders correctly", () => {
    const props = {
        title: "Song Title",
        artist: "Artist Name",
        length: "3:45",
    };

    const { container } = render(<PlayListItem {...props} />);

    // Snapshot test
    expect(container).toMatchSnapshot();

    // Extra assertions to verify props are rendered correctly
    expect(screen.getByText("Song Title")).toBeTruthy();
    expect(screen.getByText("Artist Name")).toBeTruthy();
    expect(screen.getByText("3:45")).toBeTruthy();
});
