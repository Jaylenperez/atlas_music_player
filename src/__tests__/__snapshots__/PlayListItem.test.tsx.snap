// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem renders correctly 1`] = `
<div>
  <div
    class="flex items-center justify-between rounded-md border-b border-gray-900 px-4 py-2 hover:bg-indigo-900"
  >
    <div>
      <h3
        class="text-lg font-medium text-sky-400"
      >
        Song Title
      </h3>
      <p
        class="text-sm text-fuchsia-500"
      >
        Artist Name
      </p>
    </div>
    <span
      class="text-sm text-fuchsia-300"
    >
      3:45
    </span>
  </div>
</div>
`;
